config/MezaCoreExtensions.yml

```diff
@@ -10,7 +10,7 @@ list:
   # SMW - Allows in-page storage and querying of data with semantics turning Mediawiki into a Knowledge Graph Interface
   - name: Semantic MediaWiki
     composer: "mediawiki/semantic-media-wiki"
-    version: "4.0.1"
+    version: "4.1.2"
```

Comment:
Upgrade from 4.01, to 4.1.2 due to multiple enhancements, security fixes, fixed deprecations, and better compatibility with both newer PHP and newer MediaWiki core.

For example, the system was generating warnings because the hook for LinksUpdateConstructed was deprecated in 1.38

Note: we did ensure that SMW is enabled before the Flow extension is loaded in opt/.deploy-meza/Extensions.php to avoid a known incompatibility.

We also observed the [compatibility chart](https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/master/docs/COMPATIBILITY.md) to ensure that SMW+MW+PHP are all in sync.

The .smw.json file needs to be writable and managed by Ansible.

ERROR: /opt/htdocs/mediawiki/extensions/SemanticMediaWiki/.smw.json is not writable. The "smwgConfigFileDir" setting should point to a directory that is persistent and writable!

I put the config in the jinja template for LocalSettings.php